cmds/dhclient/dhclient.go

```diff
@@ -46,17 +47,27 @@ var (
 )
 
 func ifup(ifname string) (netlink.Link, error) {
-	iface, err := netlink.LinkByName(ifname)
-	if err != nil {
-		return nil, fmt.Errorf("cannot get interface by name %v: %v", ifname, err)
-	}
-
-	if err := netlink.LinkSetUp(iface); err != nil {
-		return nil, fmt.Errorf("%v: %v can't make it up: %v", ifname, iface, err)
-	}
+	start := time.Now().Second()
```

Comment:
I tested the Now.Second method on my mac (go playground time does not work) and this only gives back a number from 0-59 (seconds in a minute) and then loops back. This is problematic for the for loop below if it started at 59 seconds and now will loop forever. An absolute value might help.